### DISCUSSION TITLE
homing docs

### DISCUSSION THREAD

**Discussion by tcurdt** [grblHAL/core#409] [Q&A] [answered]
I am a bit lost on the options for homing.

Here is what I have configured for the moment

![Screenshot 2023-12-19 at 22 07 10](https://github.com/grblHAL/core/assets/13697/c1771517-6283-42d4-8a41-bc1452bc05b9)

It seems by default I still have to home until I can move any motors.
Is this GRBHAL or maybe UGS?

What I would want ideally:
- require homing on start
- let me override the need for homing through an explicit action and let me move axis

It sounds like I want:
- homing on startup required
- override locks

Not sure how the override works though.

Is there some documentation on the options that I didn't find yet?

**Comment by terjeio** [comment]
> Not sure how the override works though.

A soft reset (sending CTRL-X or pressing the Reset button in ioSender) should work. Have you tried that?

> Is there some documentation on the options that I didn't find yet?

I have not written any - $$=22 returns the description as shown in the ioSender image above:
```

$22: Homing cycle as bitfield where setting bit 0 enables the rest:
    0 - Enable (1)
    1 - Enable single axis commands (2)
    2 - Homing on startup required (4)
    3 - Set machine origin to 0 (8)
    4 - Two switches shares one input pin (16)
    5 - Allow manual (32)
    6 - Override locks (64)
    7 - Keep homed status on reset (128)

Enables homing cycle. Requires limit switches on axes to be automatically homed.

When `Enable single axis commands` is checked, single axis homing can be performed by $H<axis letter> commands.

When `Allow manual` is checked, axes not homed automatically may be homed manually by $H or $H<axis letter> commands.

`Override locks` is for allowing a soft reset to disable `Homing on startup required`.
ok
```

**Reply by tcurdt** [reply]
> A soft reset (sending CTRL-X or pressing the Reset button in ioSender) should work. Have you tried that?

Unfortunately I cannot really use ioSender :-(

I have it somewhat running under emulation but only use it to configure GRBLHAL.
Not sure I trust that setup enough for an actual job.

When I use UGS and do a soft reset it shows

```
GrblHAL 1.1f ['$' or '$HELP' for help]
>>> $$
...
```

I assume that's considered a soft reset?

For now I have set:

![Screenshot 2023-12-20 at 13 00 28](https://github.com/grblHAL/core/assets/13697/e3bcd85b-0b1b-42ec-b784-996e1e356fa9)

but that seems to disable homing in UGS:

"homing must be enabled in the firmware settings"

...and I don't see what GRBLHAL setting that is.

---

Could I maybe just trigger the homing via console? Similar to `$HX`?
Is there a `$H` command?

I was looking for here for all supported commands https://github.com/grblHAL/core#supported-g-codes but that didn't answer that question.

**Reply by terjeio** [reply, marked as answer]
> "homing must be enabled in the firmware settings"
...and I don't see what GRBLHAL setting that is.

It is $22, same as Grbl but with additional options (it is a bitmap in grblHAL). If UGS detect homing enabled by checking for a value of 1 it will fail if options are selected, it should check for a value > 0,

> Is there a $H command?

Yes, allowed when $22 has a value > 0.

> I was looking for here for all supported commands https://github.com/grblHAL/core#supported-g-codes but that didn't answer that question.

$-prefixed commands are system commands specific to Grbl/grblHAL, not gcode... Send `$help commands` to output a list of available commands.

**Reply by tcurdt** [reply]
Perfect! `$H` does the trick.
And the override seems to work with the above settings. Thanks!
